pkg/zabbix_api.go

```diff
@@ -275,3 +280,336 @@ func isNotAuthorized(message string) bool {
 		message == "Not authorised." ||
 		message == "Not authorized."
 }
+
+func (ds *ZabbixDatasource) queryNumericItems(ctx context.Context, tsdbReq *datasource.DatasourceRequest) (*datasource.DatasourceResponse, error) {
+	jsonQueries := make([]*simplejson.Json, 0)
+	for _, query := range tsdbReq.Queries {
+		json, err := simplejson.NewJson([]byte(query.ModelJson))
+		if err != nil {
+			return nil, err
+		}
+
+		jsonQueries = append(jsonQueries, json)
+	}
+
+	if len(jsonQueries) == 0 {
+		return nil, errors.New("At least one query should be provided")
+	}
+
+	firstQuery := jsonQueries[0]
+
+	groupFilter := firstQuery.GetPath("group", "filter").MustString()
+	hostFilter := firstQuery.GetPath("host", "filter").MustString()
+	appFilter := firstQuery.GetPath("application", "filter").MustString()
+	itemFilter := firstQuery.GetPath("item", "filter").MustString()
+
+	items, err := ds.getItems(ctx, tsdbReq.GetDatasource(), groupFilter, hostFilter, appFilter, itemFilter, "num")
```

Comment:
`getItems()` return empty list in my case. Looks like something goes wrong there.

Reply:
We're able to get items with specific queries. Could this be related to the Regex filter not working properly?

Reply:
If any regex was used then it would never return any items, with the latest change that should be resolved.